lib/galaxy/jobs/runners/util/pykube_util.py

```diff
@@ -97,13 +97,17 @@ def stop_job(job, cleanup="always"):
     if not api_delete and cleanup == "onsuccess" and not job_failed:
         api_delete = True
     if api_delete:
-        delete_options = {
-            "apiVersion": "v1",
-            "kind": "DeleteOptions",
-            "propagationPolicy": "Background"
-        }
-        r = job.api.delete(json=delete_options, **job.api_kwargs())
-        job.api.raise_for_status(r)
+        if use_job_ttl is None:
+            delete_options = {
+                "apiVersion": "v1",
+                "kind": "DeleteOptions",
+                "propagationPolicy": "Background"
+            }
+            r = job.api.delete(json=delete_options, **job.api_kwargs())
+            job.api.raise_for_status(r)
+        else:
```

Comment:
What happens when the user requests cancellation of the job (via the API or by deleting all outputs) ?
Will this only happen when the ttl is reached ?

Reply:
I think you're right, this would cause the job to not be cleaned up till it runs to completion, even if the user requests cancellation. 

The original reason for doing this was to be able to inspect recently completed containers, without retaining them indefinitely, but I now feel that this no longer serves a purpose.

I guess we can just revert the stop_job function to what it was, and just send the `ttlSecondsAfterFinished` as part of the job spec anyway. This would provide an additional layer of confidence that clean up will occur, but cleanup will usually be more or less be immediate (depending on the policy).

I'll issue a new PR for this.

Reply:
Fixed in: https://github.com/galaxyproject/galaxy/pull/9784. Thanks for flagging this @mvdbeek

Reply:
Just noting that specifying `cleanup="never"`along with `ttlSecondsAfterFinished` should restore similar behaviour.